### PR TITLE
fixed the error when using stdlib in issue#100

### PR DESCRIPTION
**Pull request by cwcai633**
Now users can use the standard libraries implemented in Skulpt (https://github.com/skulpt/skulpt/tree/93d0dc6f19ba2008716787f737f2003cf6ea5bb9/src/lib, the js files).

A screenshot:
screenshot from 2017-04-18 11 26 31

### CODE REVIEW

File: client/domain/CodeSubmissionObjectFactory.js [hunk 1]
```diff
@@ -74,6 +74,21 @@ tie.factory('CodeSubmissionObjectFactory', [
       }
     };
 
+    CodeSubmission.prototype.advanceImports = function() {
+      var pattern = new RegExp('^\\ {4}import\\ \\w+$');
```
Comment (eyurko) [inline client/domain/CodeSubmissionObjectFactory.js]: Make this regex a constant somewhere in the file.

File: client/domain/CodeSubmissionObjectFactory.js [hunk 2]
```diff
@@ -74,6 +74,21 @@ tie.factory('CodeSubmissionObjectFactory', [
...
+      var insertPos = 0;
+      for (var i = 0; i < this._preprocessedCodeLines.length; i++) {
+        if (pattern.test(this._preprocessedCodeLines[i])) {
+          var importLine = this._preprocessedCodeLines[i].slice(4);
```
Comment (eyurko) [inline client/domain/CodeSubmissionObjectFactory.js]: Could you make the numbers here constants of some kind? It's not clear what this code is doing just from looking at it, and that will cause us readability issues down the line.

File: client/domain/CodeSubmissionObjectFactory.js (hunk 1 above)
Comment (cwcai633) [inline client/domain/CodeSubmissionObjectFactory.js]: Done.

File: client/domain/CodeSubmissionObjectFactory.js (hunk 2 above)
Comment (cwcai633) [inline client/domain/CodeSubmissionObjectFactory.js]: Done.

File: client/domain/CodeSubmissionObjectFactory.js [hunk 3]
```diff
@@ -34,6 +34,9 @@ tie.factory('CodeSubmissionObjectFactory', [
       }
     };
 
+    CodeSubmission.prototype.IMPORT_PATTERN = new RegExp('^\\ {4}import\\ \\w+$');
+    CodeSubmission.prototype.START_INDENT = '    ';
```
Comment (eyurko) [inline client/domain/CodeSubmissionObjectFactory.js]: I'd just set this as an int called "TAB_LENGTH"

File: client/domain/CodeSubmissionObjectFactory.js [hunk 4]
```diff
@@ -74,6 +77,23 @@ tie.factory('CodeSubmissionObjectFactory', [
       }
     };
 
+    // Move all global imports in student code out of StudentCode
+    // class
+    CodeSubmission.prototype.advanceImports = function() {
```
Comment (eyurko) [inline client/domain/CodeSubmissionObjectFactory.js]: removeImportsFromStudentCode

File: client/domain/CodeSubmissionObjectFactory.js (hunk 3 above)
Comment (cwcai633) [inline client/domain/CodeSubmissionObjectFactory.js]: Done.

File: client/domain/CodeSubmissionObjectFactory.js (hunk 4 above)
Comment (cwcai633) [inline client/domain/CodeSubmissionObjectFactory.js]: Done.

File: client/domain/CodeSubmissionObjectFactory.js [hunk 5]
```diff
@@ -34,6 +34,9 @@ tie.factory('CodeSubmissionObjectFactory', [
       }
     };
 
+    CodeSubmission.prototype.IMPORT_PATTERN = new RegExp('^\\ {4}import\\ \\w+$');
+    CodeSubmission.prototype.TAB_LENGHT = 4;
```
Comment (eyurko) [inline client/domain/CodeSubmissionObjectFactory.js]: LENGTH

Comment (cwcai633) [inline client/domain/CodeSubmissionObjectFactory.js]: Done.

**Pull request merged**